packages/instantsearch.js/src/components/Template/Template.tsx

```diff
@@ -21,7 +33,7 @@ const defaultProps = {
 export type TemplateProps = {
   data?: Record<string, any>;
   rootProps?: Record<string, any>;
-  rootTagName?: keyof JSX.IntrinsicElements;
+  rootTagName: keyof JSX.IntrinsicElements | 'fragment';
```

Comment:
```suggestion
  rootTagName: keyof JSX.IntrinsicElements | Fragment;
```

I can get this working if I change the defaultProps FYI. I don' think it matters much, but in case you want it:

```diff
diff --git a/packages/instantsearch.js/src/components/Template/Template.tsx b/packages/instantsearch.js/src/components/Template/Template.tsx
index 796dfd127..70f2531d9 100644
--- a/packages/instantsearch.js/src/components/Template/Template.tsx
+++ b/packages/instantsearch.js/src/components/Template/Template.tsx
@@ -22,9 +22,11 @@ class RawHtml extends Component<{ content: string }> {
   }
 }
 
+type TagName = keyof JSX.IntrinsicElements | typeof Fragment;
+
 const defaultProps = {
   data: {},
-  rootTagName: 'div',
+  rootTagName: 'div' as TagName,
   useCustomCompileOptions: {},
   templates: {},
   templatesConfig: {},
@@ -33,7 +35,7 @@ const defaultProps = {
 export type TemplateProps = {
   data?: Record<string, any>;
   rootProps?: Record<string, any>;
-  rootTagName: keyof JSX.IntrinsicElements | 'fragment';
+  rootTagName: TagName;
   templateKey: string;
   bindEvent?: BindEventForHits;
   sendEvent?: SendEventForHits;
@@ -69,8 +71,7 @@ See: https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/#up
       );
     }
 
-    const RootTagName =
-      this.props.rootTagName === 'fragment' ? Fragment : this.props.rootTagName;
+    const RootTagName = this.props.rootTagName;
 
     const useCustomCompileOptions =
       this.props.useCustomCompileOptions[this.props.templateKey];
diff --git a/packages/instantsearch.js/src/components/Template/__tests__/Template-test.tsx b/packages/instantsearch.js/src/components/Template/__tests__/Template-test.tsx
index ef3b246c5..f6c4603f0 100644
--- a/packages/instantsearch.js/src/components/Template/__tests__/Template-test.tsx
+++ b/packages/instantsearch.js/src/components/Template/__tests__/Template-test.tsx
@@ -5,7 +5,7 @@
 
 import { mount, shallow } from '@instantsearch/testutils/enzyme';
 import { render } from '@testing-library/preact';
-import { h } from 'preact';
+import { Fragment, h } from 'preact';
 
 import { warning } from '../../../lib/utils';
 import Template from '../Template';
@@ -58,7 +58,7 @@ describe('Template', () => {
 
   it('can have Fragment as rootTagName with string template', () => {
     const props = getProps({
-      rootTagName: 'fragment',
+      rootTagName: Fragment,
       templates: { test: '<span>test</span>' },
     });
     const wrapper = render(<Template {...props} />);
@@ -68,7 +68,7 @@ describe('Template', () => {
 
   it('can have Fragment as rootTagName with Preact template', () => {
     const props = getProps({
-      rootTagName: 'fragment',
+      rootTagName: Fragment,
       templates: { test: () => <span>test</span> },
     });
     const wrapper = render(<Template {...props} />);
```

Reply:
Ah yes I could have just narrowed it indeed thanks ! Was thinking of more radical solutions haha
But ultimately when we drop support for Hogan the whole `Template` component should go away, so it's probably best not to have to import `Fragment` in files that later won't need it